src/components/carousel/carousel.component.js

```diff
@@ -0,0 +1,366 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';
+import { compact, assign } from 'lodash';
+import { withTheme } from 'styled-components';
+import tagComponent from '../../utils/helpers/tags/tags';
+import Slide from './slide/slide.component';
+import {
+  CarouselPreviousButtonWrapperStyle,
+  CarouselNextButtonWrapperStyle,
+  CarouselButtonStyle,
+  CarouselStyledIconRight,
+  CarouselStyledIconLeft,
+  CarouselSelectorWrapperStyle,
+  CarouselSelectorInputWrapperStyle,
+  CarouselSelectorInputStyle,
+  CarouselSelectorLabelStyle,
+  CarouselWrapperStyle,
+  CarouselSliderWrapper
+} from './carousel.style';
+import { isClassic } from '../../utils/helpers/style-helper';
+import baseTheme from '../../style/themes/base';
+
+const NEXT = 'next';
+const PREVIOUS = 'previous';
+const TRANSITION_TIME = 750;
+
+class Carousel extends React.Component {
+  constructor(...args) {
+    super(...args);
+
+    /** Direction of animation */
+    this.transitionDirection = NEXT;
+
+    this.onPreviousClick = this.onPreviousClick.bind(this);
+    this.onNextClick = this.onNextClick.bind(this);
+    this.onSlideSelection = this.onSlideSelection.bind(this);
+    this.enableButtonsAfterTimeout = this.enableButtonsAfterTimeout.bind(this);
+    this.previousButtonProps = this.previousButtonProps.bind(this);
+    this.nextButtonProps = this.nextButtonProps.bind(this);
+    this.numOfSlides = this.numOfSlides.bind(this);
+    this.visibleSlide = this.visibleSlide.bind(this);
+    this.slideSelector = this.slideSelector.bind(this);
+    this.transitionName = this.transitionName.bind(this);
+  }
+
+  state = {
+    selectedSlideIndex: null, // Currently selected slide
+    disabled: false // Next/Previous buttons disabled state
+  };
+
+  /**
+   * A lifecycle method that is called after before initial render.
+   * Can set up state of component without causing a re-render
+   */
+  componentWillMount() {
+    const selectedIndex = Number(this.props.slideIndex) || Number(this.props.initialSlideIndex);
+    this.setState({ selectedSlideIndex: selectedIndex });
+  }
+
+  /** A lifecycle method that is called before re-render. */
+  componentWillReceiveProps(nextProps) {
```

Comment:
This is a legacy lifecycle method and needs updating to use `componentDidUpdate(prevProps)`